app/main/routes.py

```diff
@@ -0,0 +1,217 @@
+"""
+This script contains the routes for the different API methods.
+
+This handles the overall routing of the application.
+"""
+from flask import g, jsonify, request
+
+from . import main
+from app import errors
+from app.auth.routes import auth
+from app.decorators import json, paginate
+from app.models import BucketList, Items, User
+
+
+@main.route('/bucketlists/', methods=['GET'])
+@auth.login_required
+@paginate()
+def get_bucketlists():
+    """
+    List all the created BucketLists.
+
+    Displays a json of all the created BucketLists and the various items
+    associated with them.
+    """
+    if request.args.get('q'):
+        return BucketList.query.filter_by(created_by=g.user.user_id).filter(
+            BucketList.name.contains(request.args.get('q')))
+    else:
+        return BucketList.query.filter_by(created_by=g.user.user_id)
+
+
+@main.route('/bucketlists/<int:list_id>', methods=['GET'])
+@auth.login_required
+@json
+def get_bucketlist(list_id):
+    """
+    Get single bucket list.
+
+    Return a json of all the information as regards a particular BucketList.
+    """
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
+    if not bucketlist or bucketlist.created_by != g.user.user_id:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+    return bucketlist, 200
+
+
+@main.route('/bucketlists/', methods=['POST'], strict_slashes=False)
+@auth.login_required
+@json
+def create_bucketlist():
+    """
+    Create a new BucketList.
+
+    Add a new bucketlist and returns the bucketlist for the user to view
+    """
+    if not request.json or 'name' not in request.json:
+        return errors.bad_request("Only JSON object is accepted. Please "
+                                  "confirm that the key 'name' exists.")
+
+    user = User.query.filter_by(username=g.user.username).first().user_id
+    try:
+        bucketlist = BucketList(name=request.json.get('name'), created_by=user)
+        bucketlist.save()
+    except:
+        return errors.bad_request("An error occurred while saving. "
+                                  "Please try again.")
+    return bucketlist, 201
+
+
+@main.route('/bucketlists/<int:list_id>', methods=['PUT'])
+@auth.login_required
+@json
+def update_bucketlist(list_id):
+    """
+    Update BucketList.
+
+    Update a BucketList name.
+    """
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
+
+    if not bucketlist or bucketlist.created_by != g.user.user_id:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+
+    if not request.json:
```

Comment:
Don't you think this check should occur first? Since there will be no need querying the db if json isn't provided.